src/hooks/useUsageCalc.ts

```diff
@@ -1,92 +1,56 @@
 import { calculateMultiHTTPUsage, calculateUsage } from 'checkUsageCalc';
 
-import { Check, CheckType, UsageValues } from 'types';
-import { checkType as getCheckType } from 'utils';
+import { CalculateUsageValues, CheckType, UsageValues } from 'types';
 import { AccountingClassNames } from 'datasource/types';
 import { useCheckInfo } from 'data/useCheckInfo';
 
-const addSSL = (check: Partial<Check>, baseClass: CheckType) => {
-  if (baseClass === CheckType.HTTP) {
-    if (check.settings?.http?.tlsConfig && Object.keys(check.settings.http.tlsConfig).length > 0) {
-      return `${baseClass}_ssl`;
-    }
-  }
+export const getAccountingClass = (calcUsageValue: CalculateUsageValues): AccountingClassNames => {
+  const { basicMetricsOnly, checkType, isSSL } = calcUsageValue;
+  let accountClass: keyof typeof AccountingClassNames = checkType;
 
-  if (baseClass === CheckType.TCP) {
-    if (check.settings?.tcp?.tlsConfig && Object.keys(check.settings.tcp.tlsConfig).length > 0) {
-      return `${baseClass}_ssl`;
-    }
+  if ((checkType === CheckType.HTTP || checkType === CheckType.TCP || checkType === CheckType.GRPC) && isSSL) {
+    accountClass = `${checkType}_ssl`;
   }
 
-  return baseClass;
-};
-
-export const getAccountingClass = (check: Partial<Check>): AccountingClassNames | undefined => {
-  if (!check.settings) {
-    return;
+  if (basicMetricsOnly) {
+    accountClass = `${accountClass}_basic`;
   }
 
-  const baseClass = getCheckType(check.settings);
-  const withSSL = addSSL(check, baseClass);
-
-  const withBasic = check.basicMetricsOnly ? `${withSSL}_basic` : withSSL;
+  return AccountingClassNames[accountClass];
+};
 
-  return AccountingClassNames[withBasic as keyof typeof AccountingClassNames];
+const calcFunctionMap = {
+  multi: calculateMultiHTTPUsage,
+  other: calculateUsage,
```

Comment:
It would be useful to document how these are being calculated. I was considering if you could merge these two functions as at first glance it appears all the multi-http entry is doing is checking how many requests are being made -- then I noticed it had hard-coded values that differ from the other usage calculations but I didn't understand why.